FreeSurfer/FreeSurferPipelineNHP.sh

```diff
@@ -0,0 +1,1219 @@
+    fi
+
+    if [ ! -e "${target_volume}" ]; then
+        log_Err_Abort "Expected target_volume: ${target_volume} DOES NOT EXIST"
+    fi
+
+    mri_vol2vol_cmd="mri_vol2vol"
+    mri_vol2vol_cmd+=" --mov ${t2w_input_file}"
+    mri_vol2vol_cmd+=" --targ ${target_volume}"
+    mri_vol2vol_cmd+=" --regheader"
+    mri_vol2vol_cmd+=" --o ${t2w_output_file}"
+
+    log_Msg "Creating ${t2w_output_file} with mri_vol2vol_cmd: ${mri_vol2vol_cmd}"
+    ${mri_vol2vol_cmd}
+    return_code=$?
+    if [ "${return_code}" != 0 ]; then
+        log_Err_Abort "mri_vol2vol command failed with return code: ${return_code}"
+    fi
+
+    popd
+}
+
+#
+# Generate QC file - T1w X T2w
+#
+make_t1wxt2w_qc_file()
+{
+    local working_dir
+    local t1w_input_file
+    local t2w_input_file
+    local output_file
+    local fslmaths_cmd
+    local return_code
+
+    working_dir="${1}"
+
+    pushd "${working_dir}"
+
+    # We should already have generated the T1w_hires.nii.gz and T2w_hires.nii.gz files
+    t1w_input_file="T1w_hires.nii.gz"
+    t2w_input_file="T2w_hires.nii.gz"
+    output_file="T1wMulT2w_hires.nii.gz"
+
+    if [ ! -e "${t1w_input_file}" ]; then
+        log_Err_Abort "Expected t1w_input_file: ${t1w_input_file} DOES NOT EXIST"
+    fi
+
+    if [ ! -e "${t2w_input_file}" ]; then
+        log_Err_Abort "Expected t2w_input_file: ${t2w_input_file} DOES NOT EXIST"
+    fi
+
+    fslmaths_cmd="fslmaths"
+    fslmaths_cmd+=" ${t1w_input_file}"
+    fslmaths_cmd+=" -mul ${t2w_input_file}"
+    fslmaths_cmd+=" -sqrt ${output_file}"
+
+    log_Msg "Creating ${output_file} with fslmaths_cmd: ${fslmaths_cmd}"
+    ${fslmaths_cmd}
+    return_code=$?
+    if [ "${return_code}" != "0" ]; then
+        log_Err_Abort "fslmaths command failed with return code: ${return_code}"
+    fi
+
+    popd
+}
+
+
+T2wtoT1wFile="T2wtoT1w.mat"      # Calling this file T2wtoT1w.mat regardless of whether the input to recon-all was -T2 or -FLAIR
+OutputOrigT2wToT1w="OrigT2w2T1w" # Needs to match name used in PostFreeSurfer (N.B. "OrigT1" here refers to the T1w/T1w.nii.gz file; NOT FreeSurfer's "orig" space)
+
+# ----------------------------------------------------------------------
+log_Msg "Starting main functionality"
+# ----------------------------------------------------------------------
+
+# ----------------------------------------------------------------------
+log_Msg "Retrieve positional parameters"
+# ----------------------------------------------------------------------
+SubjectDIR="${1}"
+SubjectID="${2}"
+T1wImage="${3}"       # Irrelevant if '--existing-subject' flag is set
+T1wImageBrain="${4}"  # Irrelevant if '--existing-subject' flag is set
+T2wImage="${5}"       # Irrelevant if '--existing-subject' flag is set
+recon_all_seed="${6}"
+
+#RunMode="$p_runmode"
+#SPECIES="$p_species"
+RunMode="${RunMode:-1}"
+SPECIES="${Species}"
+
+# Species-specific environments imported from SPECIES4FS.cnf
+export SPECIES
+source ${HCPPIPEDIR}/FreeSurfer/custom/SetUpFSNHP.sh
```

Comment:
Let's make this simpler.  Also, do *not* use `newopts.shlib` in the `SetUp...` this time (its already sourced and populated with arguments since this is a pipeline, not a batch file, and I am not going to support the potential collisions for that case...but also, `newopts.shlib` is not designed for things that are sourced in general, and is overcomplicated for a small number of arguments).
```suggestion
source "$HCPPIPEDIR"/FreeSurfer/custom/SetUpFSNHP.sh "$Species"
```

Reply:
Fixed it.
819e0ffeb6a4d71d2b47cd98333e96ff5dd14f2b